lib/components/viewers/live-stop-times.js

```diff
@@ -0,0 +1,161 @@
+
+  componentWillUnmount () {
+    // Turn off auto refresh unconditionally (just in case).
+    this._stopAutoRefresh()
+  }
+
+  componentDidUpdate (prevProps) {
+    // Handle stopping or starting the auto refresh timer.
+    if (prevProps.autoRefreshStopTimes && !this.props.autoRefreshStopTimes) this._stopAutoRefresh()
+    else if (!prevProps.autoRefreshStopTimes && this.props.autoRefreshStopTimes) this._startAutoRefresh()
+  }
+
+  render () {
+    const {
+      homeTimezone,
+      stopData,
+      stopViewerArriving,
+      stopViewerConfig,
+      timeFormat,
+      transitOperators
+    } = this.props
+    const { spin } = this.state
+
+    // construct a lookup table mapping pattern (e.g. 'ROUTE_ID-HEADSIGN') to
+    // an array of stoptimes
+    const stopTimesByPattern = getStopTimesByPattern(stopData)
+    const routeComparator = coreUtils.route.makeRouteComparator(
+      transitOperators
+    )
+    const patternHeadsignComparator = coreUtils.route.makeStringValueComparator(
+      pattern => pattern.pattern.headsign
+    )
+    const patternComparator = (patternA, patternB) => {
+      // first sort by routes
+      const routeCompareValue = routeComparator(
+        patternA.route,
+        patternB.route
+      )
+      if (routeCompareValue !== 0) return routeCompareValue
+
+      // if same route, sort by headsign
+      return patternHeadsignComparator(patternA, patternB)
+    }
+
+    return (
+      <>
+        <div>
+          {Object.values(stopTimesByPattern)
+            .sort(patternComparator)
+            .map(({ id, pattern, route, times }) => {
+              // Only add pattern if route info is returned by OTP.
+              return routeIsValid(route, getRouteIdForPattern(pattern))
+                ? (
+                  <PatternRow
+                    homeTimezone={homeTimezone}
+                    key={id}
+                    pattern={pattern}
+                    route={route}
+                    stopTimes={times}
+                    stopViewerArriving={stopViewerArriving}
+                    stopViewerConfig={stopViewerConfig}
+                    timeFormat={timeFormat}
+                  />
+                )
+                : null
+            })
+          }
+        </div>
+
+        {/* Auto update controls for realtime arrivals */}
+        <div style={{ marginTop: '20px' }}>
+          {/* eslint-disable-next-line jsx-a11y/label-has-for */}
+          <label style={{ fontWeight: 300, fontSize: 'small' }}>
+            <input
+              checked={this.props.autoRefreshStopTimes}
+              name='autoUpdate'
+              onChange={this._onToggleAutoRefresh}
+              type='checkbox'
+            />{' '}Auto-refresh arrivals?
+          </label>
+          <button
+            className='link-button pull-right'
+            onClick={this._refreshStopTimes}
+            style={{ fontSize: 'small' }}
+          >
+            <Icon
+              className={spin ? 'fa-spin' : ''}
```

Comment:
This button no longer spins on click.

Reply:
Fixed in https://github.com/opentripplanner/otp-react-redux/pull/292/commits/08253eff2bb186584000d7209a5635d6de86dbfb.